idioms/priv-extend.md

```diff
@@ -1,35 +1,52 @@
-# Privacy for extensibility
+# `#[non_exhaustive]` for extensibility
 
 ## Description
 
-Use a private field to ensure that a struct is extensible without breaking
+Use `#[non_exhaustive]` on `struct`, `enum` and `enum` variant definitions to ensure that a struct is extensible without breaking
```

Comment:
enum and enum?

Reply:
"Enum variant", maybe it should be hyphenated?

Reply:
sounds good

Reply:
Updated to match how it's described in https://doc.rust-lang.org/reference/attributes/type_system.html#the-non_exhaustive-attribute